Added link to German Baikal tutorial Follow Up

JsBergbau opened the pull request:
Here is now the final Pull-Request for Baikal Tutorial
Big thanks to @ByteHamster for converting it from docx to markdown.
See original discussion here https://github.com/sabre-io/Baikal/pull/956

ByteHamster commented on README.md:
```diff
@@ -7,6 +7,8 @@ This is the source repository for the Baïkal CalDAV and CardDAV server.
 
 Head to [sabre.io/baikal][2] for information about installation, upgrading and troubleshooting.
 
+A German tutorial about installation of Baïkal, using it together with Thunderbird, Android and DavX5, sharing calendars, database maintenance and security can be found [here][6].  
```
I just had time to read the tutorial in detail. It seems to be focused a lot on the Raspberry Pi, so this should be mentioned here.

Some remarks about the content of the tutorial:
- The install section mentions that calendar sharing is not supported. I would change that paragraph to be up-to-date instead of writing something outdated with an "update". For readers, it does not really matter that there was an old version without support.
- I am confused about the paragraph about the Influxdb port. I would move it to the end of the web server configuration section. Before that, you did not tell that you plan to use port 9999
- The thing that blocks the specific folder in the nginx config should also block the config folder. This is a security risk otherwise. You root seems to be the html folder, though, so specific and config are not accessible anyway. So depending on what you want to do, please either add the config folder or remove that section of the config file because it does not apply in your case.
- Creating the nginx config in the Baikal root folder is pretty strange. It is usually created in the sites-available folder of nginx. The problem I see with having it in the Baikal folder is that the file might be overwritten when updating the Baikal folder.
- I would remove the password security calculation. This assumes that an attacker wants to find your exact password. The thing is, md5 allows to calculate "backwards" to get a password (usually not the one you chose) that has the same hash. So independently of the password, "only" 2^128 tries are needed to get in. Not sure how feasible this is. This would need some more research but the (maybe wrong) claim can be avoided by removing that paragraph :)
- I would remove the sentence that the email address is not important. It is when sharing, so nobody should be tempted to enter something wrong.
- You mix install via composer (experts only) and via zip file (without composer, for most users). This could be confusing.
- The user login section is pretty unexpected. Normal users (no sharing) do not need to log in. This should have a paragraph about when this is needed.
- clearing addressbookchanges is not advisable as is breaks sync for clients that have not been connected for a long time. I do not think that this needs a significant amount of storage.

JsBergbau commented:
>  It seems to be focused a lot on the Raspberry Pi, so this should be mentioned here.

Focus is also for people with less IT/Linux knowlegdge, thats why it is so detailed. Changed it


> The install section mentions that calendar sharing is not supported. I would change that paragraph to be up-to-date instead of writing something outdated with an "update". For readers, it does not really matter that there was an old version without support.

Good point, changed it.

> I am confused about the paragraph about the Influxdb port. I would move it to the end of the web server configuration section. Before that, you did not tell that you plan to use port 9999

I thought I place it before first occurance, so that you don't have already created webserver config and then have to change it again.

> The thing that blocks the specific folder in the nginx config should also block the config folder. This is a security risk otherwise. You root seems to be the html folder, though, so specific and config are not accessible anyway. So depending on what you want to do, please either add the config folder or remove that section of the config file because it does not apply in your case.

Can you explain a bit more?
We're in Baikal root folder and change everything to www-data so including the confing folder.
Only the nginx config file gets root as owner
It looks then like
![grafik](https://user-images.githubusercontent.com/37013344/88483474-b68cb300-cf57-11ea-8273-dc3cc82f480c.png)

> Creating the nginx config in the Baikal root folder is pretty strange. It is usually created in the sites-available folder of nginx. The problem I see with having it in the Baikal folder is that the file might be overwritten when updating the Baikal folder.

It allows better maintainance and backup. You have only to copy the Baikal folder and everything is backed up since it is only linked to the sites-available folder. Thats why I like this way to have everything in one place.

> I would remove the password security calculation. This assumes that an attacker wants to find your exact password. The thing is, md5 allows to calculate "backwards" to get a password (usually not the one you chose) that has the same hash. So independently of the password, "only" 2^128 tries are needed to get in. Not sure how feasible this is. This would need some more research but the (maybe wrong) claim can be avoided by removing that paragraph :)

Thanks for the hint. Very good point. I've done some reasearch and updated the paragraph. Maximum PW length with leads to no more security is 22 in this case.

> I would remove the sentence that the email address is not important. It is when sharing, so nobody should be tempted to enter something wrong.

Can you explain more? There is never send any E-Mail to this address and you need, as explained, this address is only needed for sharing the calendar and you just have to type exactly whats written there. 

> You mix install via composer (experts only) and via zip file (without composer, for most users). This could be confusing.

Install is via composer, update via zip. Since there is explained everything step by step, I don't see so much confusion. But lets see when the first issues are opened.

> The user login section is pretty unexpected. Normal users (no sharing) do not need to log in. This should have a paragraph about when this is needed.

Thanks, Added it.

> clearing addressbookchanges is not advisable as is breaks sync for clients that have not been connected for a long time. I do not think that this needs a significant amount of storage.
Thanks for the hint.  Can you explain briefly how these entries help in syncing process? I thought these are just some kind of log entries. I've updated this section with a warning and perhaps I'll even delete it.

ByteHamster commented:
> I thought I place it before first occurance, so that you don't have already created webserver config and then have to change it again.

Then please add a sentence before that telling that you plan to set up port 9999

> > The thing that blocks the specific folder in the nginx config should also block the config folder. This is a security risk otherwise. You root seems to be the html folder, though, so specific and config are not accessible anyway. So depending on what you want to do, please either add the config folder or remove that section of the config file because it does not apply in your case.
> 
> Can you explain a bit more?

There are two basic set-ups most users seem to use.

- Push the whole Baikal directory to a web server and access using `https://domain.tld/baikal/html/dav.php`. In that case, someone could write `https://domain.tld/baikal/config/baikal.yaml` to get access to the database passwords. The ngnix config needs something like `location ~ /(.ht|Core|Specific|config) { deny all;`. Your example does not contain the `config` part.
- Create a vhost that allows to access using `https://domain.tld/dav.php`. There is no way to address the Baikal config file, now. So the ngnix config does not need the blocking part.

Your config seems to be a mix of both methods. Blocking access (but leaving the config folder open) **and** using a direct vhost.

> Can you explain more? There is never send any E-Mail to this address and you need, as explained, this address is only needed for sharing the calendar and you just have to type exactly whats written there.

I think there can be when inviting users to an event (not a calendar).

> Can you explain briefly how these entries help in syncing process? I thought these are just some kind of log entries. I've updated this section with a warning and perhaps I'll even delete it.

They are very important to sync. When asking the server for database changes since the last sync, it will look up in that table. The changes are enumerated, so every client just needs to save the version ID (`synctoken`) of its previous sync. Deleting rows in that table can lead to nasty sync issues that require deleting the account from devices and re-adding it to force a full sync with the new `synctoken`

JsBergbau commented:
Hi ByteHamster,

first thanks for all your input, that improves the Baikal tutorial a lot

> Then please add a sentence before that telling that you plan to set up port 9999
Done.

> There are two basic set-ups most users seem to use.
> 
>     Push the whole Baikal directory to a web server and access using https://domain.tld/baikal/html/dav.php. In that case, someone could write https://domain.tld/baikal/config/baikal.yaml to get access to the database passwords. The ngnix config needs something like location ~ /(.ht|Core|Specific|config) { deny all;. Your example does not contain the config part.
>     Create a vhost that allows to access using https://domain.tld/dav.php. There is no way to address the Baikal config file, now. So the ngnix config does not need the blocking part.
> 
> Your config seems to be a mix of both methods. Blocking access (but leaving the config folder open) and using a direct vhost.

Thanks. Now I've understood. I've added the config block and a comment about that sections that it is only for reference there.

> I think there can be when inviting users to an event (not a calendar).
Yeah probably when you've setup a "Email invite sender address" which isn't done in this tutorial. Nevertheless I've mentioned that correct E-Mail-Adresses are required when configuring "Email invite sender address".

> They are very important to sync. When asking the server for database changes since the last sync, it will look up in that table. The changes are enumerated, so every client just needs to save the version ID (synctoken) of its previous sync. Deleting rows in that table can lead to nasty sync issues that require deleting the account from devices and re-adding it to force a full sync with the new synctoken

Thanks for this import info. I've rewritten the paragraph and recommended database maintance only to expert users and that only executing VACUUM is safe.

JsBergbau commented:
Hi @ByteHamster ,
just wanted to know if Tutorial is now ready that the link to it can be merged in Baikal ReadMe

ByteHamster commented:
The text on GitHub looks good to me, now. What do you think, @phil-davis @DeepDiver1975?

JsBergbau commented:
Hi @ByteHamster,

about 1 month has passed since updating/changing the tutorial. Do you like some additional changes or is it ready to be linked?

phil-davis commented:
@ByteHamster @JsBergbau I am not a German speaker, so I can't give a useful opinion.

ByteHamster commented on README.md:
```diff
@@ -7,6 +7,8 @@ This is the source repository for the Baïkal CalDAV and CardDAV server.
 
 Head to [sabre.io/baikal][2] for information about installation, upgrading and troubleshooting.
 
+A German tutorial about installation of Baïkal, using it together with Thunderbird, Android and DavX5, sharing calendars, database maintenance and security can be found [here][6]. It's focus is for people with not so much IT experience. Therefore it is very detailed, step by step with a RaspberryPI used as server.
```
I think this should be `Its`, not `It's`. `It's` is a short form for `It is`, if I remember correctly.

JsBergbau commented:
Thanks for paying attention. You're correct with `it's and its`. I have corrected it.

ByteHamster commented:
Sorry for taking so long to reply. I am currently pretty busy with other projects.

Thank you very much for taking the time to write the very detailed tutorial!
